File: android-app/app/src/main/java/com/petdocs/android/ui/theme/Theme.kt
package com.petdocs.android.ui.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Shapes
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * Petdocs brand palette.
 *
 * - Background cream #FFFBF5, primary teal #0D9488, accent amber #F59E0B,
 *   text ink #1C1917. Surface is white.
 * - Display font is the platform default (no downloadable fonts) so the app
 *   stays fully usable offline.
 */
object PetdocsColors {
    val Cream = Color(0xFFFFFBF5)
    val CreamDark = Color(0xFFF5EBDD)
    val Teal = Color(0xFF0D9488)
    val TealDark = Color(0xFF0B7C72)
    val TealContainer = Color(0xFFCCFBF1)
    val OnTealContainer = Color(0xFF134E4A)
    val Amber = Color(0xFFF59E0B)
    val AmberDark = Color(0xFFB45309)
    val AmberContainer = Color(0xFFFEF3C7)
    val OnAmberContainer = Color(0xFF78350F)
    val Ink = Color(0xFF1C1917)
    val InkSoft = Color(0xFF78716C)
    val White = Color(0xFFFFFFFF)
    val Danger = Color(0xFFDC2626)
    val Success = Color(0xFF16A34A)
}

private val PetdocsLightColors = lightColorScheme(
    primary = PetdocsColors.Teal,
    onPrimary = PetdocsColors.White,
    primaryContainer = PetdocsColors.TealContainer,
    onPrimaryContainer = PetdocsColors.OnTealContainer,
    secondary = PetdocsColors.Amber,
    onSecondary = PetdocsColors.Ink,
    secondaryContainer = PetdocsColors.AmberContainer,
    onSecondaryContainer = PetdocsColors.OnAmberContainer,
    tertiary = PetdocsColors.AmberDark,
    background = PetdocsColors.Cream,
    onBackground = PetdocsColors.Ink,
    surface = PetdocsColors.White,
    onSurface = PetdocsColors.Ink,
    surfaceVariant = PetdocsColors.CreamDark,
    onSurfaceVariant = PetdocsColors.InkSoft,
    outline = PetdocsColors.InkSoft,
    error = PetdocsColors.Danger,
    onError = PetdocsColors.White,
)

/** Default Material3 type scale — platform font only (offline-safe, no downloads). */
private val PetdocsTypography = Typography()

/** Rounded 16dp geometry used across cards, sheets, and dialogs. */
private val PetdocsShapes = Shapes(
    small = RoundedCornerShape(12.dp),
    medium = RoundedCornerShape(16.dp),
    large = RoundedCornerShape(16.dp),
    extraLarge = RoundedCornerShape(16.dp),
)

/**
 * Petdocs Material3 theme (light). Imported by `MainActivity` as
 * `com.petdocs.android.ui.theme.PetdocsTheme`.
 */
@Composable
fun PetdocsTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = PetdocsLightColors,
        typography = PetdocsTypography,
        shapes = PetdocsShapes,
        content = content,
    )
}

File: ios-app/PetDocs/Theme.swift
import SwiftUI

// Petdocs brand palette. Cream background, teal primary, amber accent,
// ink text. Matches Android PetdocsColors and the web design tokens.
extension Color {
    static let cream = Color(red: 1.0, green: 0.984, blue: 0.961) // FFFBF5
    static let brandTeal = Color(red: 0.051, green: 0.580, blue: 0.533) // 0D9488
    static let brandAmber = Color(red: 0.961, green: 0.620, blue: 0.043) // F59E0B
    static let ink = Color(red: 0.110, green: 0.098, blue: 0.090) // 1C1917
    static let inkSoft = Color(red: 0.471, green: 0.443, blue: 0.424) // 78716C
}

struct CardStyle: ViewModifier {
    func body(content: Content) -> some View {
        content
            .padding(16)
            .background(Color.white)
            .cornerRadius(16)
            .shadow(color: Color.ink.opacity(0.06), radius: 8, x: 0, y: 2)
    }
}

extension View {
    // Rounded 16 card used across lists, sheets, and dialogs.
    func petCard() -> some View {
        modifier(CardStyle())
    }

    // Keeps every tappable control at least 48pt, per platform guidance.
    func minTouchTarget() -> some View {
        frame(minWidth: 48, minHeight: 48)
    }
}
